apps/src/storage/dataBrowser/DataVisualizer.jsx

```diff
@@ -56,24 +56,75 @@ class DataVisualizer extends React.Component {
 
   updateChart = () => {
     const targetDiv = document.getElementById('chart-area');
+    if (!targetDiv) {
+      return;
+    }
     const records =
       Object.keys(this.props.tableRecords).length > 0 &&
       this.props.tableRecords.map(tableRecord => JSON.parse(tableRecord));
-    if (this.state.chartType === 'Bar Chart' && this.state.values) {
-      var chart = new GoogleChart.MaterialBarChart(targetDiv);
-      const chartData = this.aggregateRecordsByColumn(
-        records,
-        this.state.values
-      );
-      chart.drawChart(chartData, [this.state.values, 'count']);
+
+    let chart;
+    let chartData;
+    const columns = [];
+    const options = {};
+    switch (this.state.chartType) {
+      case 'Bar Chart':
+        if (this.state.values) {
+          chart = new GoogleChart.MaterialBarChart(targetDiv);
+          chartData = this.aggregateRecordsByColumn(records, this.state.values);
+          columns.push(this.state.values, 'count');
+        }
+        break;
+      case 'Histogram':
+        if (this.state.values && this.state.bucketSize) {
+          let min = Infinity;
+          let max = -Infinity;
+          records.forEach(record => {
+            let value = record[this.state.values];
+            if (value < min) {
+              min = value;
+            }
+            if (value > max) {
+              max = value;
+            }
+          });
+          options.histogram = {
+            bucketSize: this.state.bucketSize
+          };
+          chart = new GoogleChart.Histogram(targetDiv);
+
+          // Only use non-empty records.
+          chartData = records.filter(record => !!record);
```

Comment:
This seems like the sort of business logic we should have a unit test for.

Reply:
Whoops no longer needed.

Reply:
I'm working on a refactoring this logic right now, and I will add a test as part of that.